### PR TITLE
Feat: Improved and Implemented UI for DM View between two users.

### PR DESCRIPTION
**Pull request by YenumJay**
This implements the Improved UI design for DM View between two Users and the codes were refactored to enable scalable implementation for the next steps of this workflow(Send Messages In DMs).

Currently, the Messages in DM are run on Simulation.

This addresses this issue #739 

DM VIEW

### CODE REVIEW

File: lib/ui/nav_pages/dm_page/dm_page_viewmodel.dart [hunk 1]
```diff
@@ -1,3 +1,11 @@
+import 'package:hng/app/app.locator.dart';
+import 'package:hng/app/app.router.dart';
 import 'package:stacked/stacked.dart';
+import 'package:stacked_services/stacked_services.dart';
 
-class DmPageViewModel extends BaseViewModel {}
+class DmPageViewModel extends BaseViewModel {
+
+  void navigateToDmUser() {
+    locator<NavigationService>().navigateTo(Routes.dmUserView);
```
Comment (isaacadariku) [inline lib/ui/nav_pages/dm_page/dm_page_viewmodel.dart:9]: Can kindly make the `NavigationService()` a variable to be use.

`final _navigationService = locator<NavigationService>();`

File: lib/ui/nav_pages/dm_page/widgets/dmmessage_read.dart [hunk 2]
```diff
@@ -9,7 +12,9 @@ class DMMessageRead extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return InkWell(
-      onTap: () {},
+      onTap: () {
+        locator<NavigationService>().navigateTo(Routes.dmUserView);
```
Comment (isaacadariku) [inline lib/ui/nav_pages/dm_page/widgets/dmmessage_read.dart:16]: Move this logic away from the view, and put in the viewmodel, also follow the above request

File: lib/ui/nav_pages/dm_page/widgets/dmmessage_unread.dart [hunk 3]
```diff
@@ -10,7 +13,9 @@ class DMMessageUnread extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return InkWell(
-      onTap: () {},
+      onTap: () {
+        locator<NavigationService>().navigateTo(Routes.dmUserView);
```
Comment (isaacadariku) [inline lib/ui/nav_pages/dm_page/widgets/dmmessage_unread.dart:17]: Here too

File: lib/ui/shared/styles.dart [hunk 4]
```diff
@@ -125,6 +125,22 @@ class AppTextStyles {
     );
   }
 
+  //status icon
+  static Widget statusIcon(int count) {
```
Comment (isaacadariku) [inline lib/ui/shared/styles.dart:129]: Can you remove the static keyword here and make the widget in a class, functional widgets are bad practice in flutter.

File: lib/ui/view/dm_user/dm_user_viewmodel.dart [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import 'dart:math';
 
 import 'package:flutter/material.dart';
+import 'package:hng/ui/view/dm_user/widgets/custom_status.dart';
```
Comment (isaacadariku) [inline lib/ui/view/dm_user/dm_user_viewmodel.dart:4]: No import of UI in the viewmodel

**Review by isaacadariku — CHANGES_REQUESTED** [review]
Good, make the few changes requested and we are good to go.

File: lib/ui/view/dm_user/widgets/group_separator.dart [hunk 6]
```diff
@@ -0,0 +1,29 @@
+
+import 'package:flutter/material.dart';
+
+Widget groupSeparator(String value) {
```
Comment (isaacadariku) [inline lib/ui/view/dm_user/widgets/group_separator.dart:4]: I spoke about functional widgets are not allowed, ensure it have it's class

**Review by isaacadariku — CHANGES_REQUESTED** [review]
Few changes and we good

**Review by mariamadebolahamzat — APPROVED** [review]
All reviews has been resolved..

**Review by isaacadariku — APPROVED** [review]
LGTM

**Pull request merged**